pkgs/applications/misc/collision/default.nix

```diff
@@ -0,0 +1,107 @@
+{ lib
+, fetchFromGitHub
+, crystal
+, libadwaita
+, gtk3
+, gobject-introspection
+, openssl
+, libxml2
+, pkg-config
+, wrapGAppsHook4
+}:
+let
+  gi_crystal = crystal.buildCrystalPackage {
+    name = "gi-crystal";
+
+    src = fetchFromGitHub {
+      owner = "hugopl";
+      repo = "gi-crystal";
+      rev = "refs/tags/v0.16.0";
+      hash = "sha256-ij4U8BoSNHpkh5SmvoH5anGB/ZdvmC5zDwJCusR/s/c=";
+    };
+
+    format = "make";
+    shardsFile = ./gi-crystal/gi-crystal-shards.nix;
+    lockFile = ./gi-crystal/gi-crystal-shard.lock;
+    patches = [./gi-crystal/gi-crystal.patch];
+
+    nativeBuildInputs = [ gobject-introspection ];
+    buildInputs = [ libadwaita ];
+
+    doCheck = false;
+    doInstallCheck = false;
+
+    buildPhase = ''
+      cd lib
+      for dep in *; do
+        if [ -d "$dep" ]; then
```

Comment:
```suggestion
      for dep in */; do
```

Reply:
Added in https://github.com/NixOS/nixpkgs/commit/9c1cd513a09b5380ac72043e690390c54590a6bb